### DISCUSSION TITLE
Indirect objects

### DISCUSSION THREAD

**Discussion by m-holger** [qpdf/qpdf-dev#13] [General]
The 2020 spec states

> NOTE
(2020) Any object outside of an object stream (see 7.5.7, "Object streams") can consist solely of
an object reference. PDF syntax thus permits chains of such objects.

Presumably qpdf should support such objects?

The errata to the spec state

> A PDF object number is a positive (non-zero) decimal integer comprised only of digits. It shall not have a leading PLUS SIGN ("+", 2Bh) and shall not start with leading zeros ("0").

> A PDF generation number is a non-negative decimal integer: its syntax requirements are identical to those of a PDF object number (above), except that the single digit "0" shall also be permitted. 

Should qpdf check for this?

**Comment by jberkenbilt** [comment]
> > A PDF generation number is a non-negative decimal integer: its syntax requirements are identical to those of a PDF object number (above), except that the single digit "0" shall also be permitted.
> 
> Should qpdf check for this?

I'd probably not. Somewhere out there, there could be a file that has 0.0 or something as a generation, and if that happened and another reader accepted it, the only sensible thing for us to do would be to accept it as well. We could warn if if we thought it was worth it.
